xarray/backends/plugins.py

```diff
@@ -70,10 +70,12 @@ def set_missing_parameters(backend_entrypoints):
 
 
 def sort_backends(backend_entrypoints):
-    ordered_backends_entrypoints = {}
-    for be_name in STANDARD_BACKENDS_ORDER:
-        if backend_entrypoints.get(be_name, None) is not None:
-            ordered_backends_entrypoints[be_name] = backend_entrypoints.pop(be_name)
+    ordered_backends_entrypoints = {
+        be_name: backend_entrypoints.pop(be_name)
+        for be_name in STANDARD_BACKENDS_ORDER
+        if be_name in backend_entrypoints
```

Comment:
are you sure this is even doing the same check? I'm guessing the convoluted `get(be_name, None) is not None` bit was for a reason...

Reply:
it seems my mistake was to assume that entries to `backend_entrypoints` are never equal to `None`. I'm not sure why that would happen, though (maybe `Entrypoint.load` returns `None` if it fails? But couldn't we just not assign then?). If that's not a mistake I would probably remove all `None` valued entries before this because it's much easier to understand if we don't do multiple things at the same time.

Reply:
I reverted back to the original version so as to avoid accidentally breaking things :)